app/registration/api.py

```diff
@@ -44,7 +44,50 @@ def api_register():
         data = registration.blob,
     )
 
-@registration_blueprint.route('/api/order/token')
+@registration_blueprint.route('/api/priorities', methods=['GET', 'POST'])
+@oauth.require_oauth('registration_priorities')
 @token_auth.login_required
-def validate_token():
-    return jsonify(g.uni.name)
+def api_registration_priorities():
+    if request.method == 'POST' \
+        and request.headers.get('Content-Type') == 'application/json':
+        req = request.get_json()
+        if not req:
+            abort(403)
+            return ''
+
+        new_priority = len(req['priority'])
+
+        registrations = Registration.query.filter_by(uni_id = g.uni.id)
+        for registration in registrations:
```

Comment:
I have no problem to use your solution, IMHO this is more a question of style ;)

Anyway, at the moment we do not have any information about guaranteed places, I do not like the idea to symbolize them with negative priorities...